### PR TITLE
Feat/cc 128 template list filter and pagination

### PR DESCRIPTION
**Pull request by Cecilia0109**
1、add infinite scroll pagination in the template list
2、add court category filter in the template list
3、when the user clicks the option, the sidebar will display the selected court category.
25251668257812_ pic

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { getCourtSpecData, CourtSpecMapper } from "@/store/reducer/courtSpecData
 import { useGetCourtsQuery } from "@/redux/api/courtSizeApi";
 import { courtSpecMapping } from "@/utils/courtSpecMapping";
 import LoadingPage from "@/components/LoadingPage";
+import { useGetTemplateListsQuery, useGetTemplatesQuery } from "@/redux/api/templateApi";
```
Comment (STY1997) [inline src/pages/index.tsx:15]: please remove

File: src/components/TemplateList/TemplateItem.tsx [hunk 2]
```diff
@@ -125,7 +114,44 @@ const TemplateItem = (prop: Props) => {
...
+      width="300px"
+      height="240px"
+      marginBottom="16px"
+      flexDirection="column"
+      alignItems="center"
+      padding="5px 10px 10px 10px"
+      backgroundColor="white"
+      opacity={templateOpacityDrop ? "0.4" : "1"}
+      onMouseEnter={selectTemplate}
+      onMouseLeave={resetHeightLight}
```
Comment (STY1997) [inline src/components/TemplateList/TemplateItem.tsx:132]: ref={ref ? ref : null}

**Comment by STY1997** [comment]
> 1、add infinite scroll pagination in the template list 2、add court category filter in the template list 3、when the user clicks the option, the sidebar will display the selected court category. 25251668257812_ pic

**Comment by STY1997** [comment]
it is better to have a space between the select bar and content

File: src/components/TemplateList/TemplateItem.tsx (hunk 2 above)
Comment (Cecilia0109) [reply src/components/TemplateList/TemplateItem.tsx]: > ref={ref ? ref : null}

Thanks  fixed

**Review by ZhihanChenZack — APPROVED** [review]

**Review by tatoMa — APPROVED** [review]
LGTM

**Pull request merged**